tests/foreman/cli/test_role.py

```diff
@@ -426,81 +431,832 @@ def test_positive_update_role_taxonomies(self):
+        its own taxonomies are not same as its role
+
+        :id: 2c0b6e8e-c8b7-4212-af79-d329bd803558
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create user with different taxonomies than above Org Admin
+                taxonomies
+            3. Assign above cloned role to user
+            4. Login with user and attempt to access resources
+
+        :expectedresults: User should not be able to access any resources and
+            permissions in taxonomies selected in Org Admin role
+
+        :CaseLevel: System
+        """
+
+    @stubbed
+    @tier3
+    def test_negative_access_entities_from_user(self):
+        """User can not access resources within its own taxonomies if assigned
+        role does not have permissions for user taxonomies
+
+        :id: 512d2758-2ca0-49c2-b80e-f8a7bffd35b4
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create user with different taxonomies than above Org Admin
+                taxonomies
+            3. Assign above cloned role to user.
+            4. Login with user and attempt to access resources
+
+        :expectedresults: User should not be able to access any resources and
+            permissions in his own taxonomies
+
+        :CaseLevel: System
         """
 
     @stubbed
     @tier2
-    def test_positive_access_resources_from_role_taxonomies(self):
-        """Test user access resources from taxonomies of assigned role
+    def test_positive_override_cloned_role_filter(self):
+        """Cloned role filter overrides
 
-        :id: 754cf23e-43ed-4d2b-b4ec-dacc77bc009d
+        :id: 0711541f-1af6-4493-b1f2-552367541d99
 
         :steps:
 
-            1. Create role with taxonomies
-            2. Create resource(s) filter(s) without overriding them
-            3. Create user with taxonomies same as role taxonomies
-            4. Assign step 1 role to user
+            1. Create a role with overridden filter
+            2. Clone above role
+            3. Attempt to override the filter in cloned role
 
-        :expectedresults: User should be able to access the resource(s) of the
-            assigned role
+        :expectedresults: Filter in cloned role should be overriding
 
-        :caseautomation: notautomated
+        :CaseLevel: Integration
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_emptiness_of_filter_taxonomies_on_role_clone(self):
+        """Taxonomies of filters in cloned role are set to None for filters that
+        are overridden in parent role
+
+        :id: 20179b43-9db7-4af4-beca-fecc7ff7490c
+
+        :steps:
+
+            1. Create a role with an overridden filter
+            2. Overridden filter should have taxonomies assigned
+            3. Clone above role
+            4. View cloned role filters
+
+        :expectedresults:
+
+            1. Taxonomies of the 'parent role overridden filters' are set to
```

Comment:
so just use it. I meant `filters that are overridden in parent role of cloned role` instead `'parent role overridden filters'`